File: Python/백준/Silver/4673. 셀프 넘버/셀프 넘버.py
# 셀프넘버

def make_num(n):
    new = n
    while True:
        if n // 10 >= 1:
            new += n % 10 
            n = n // 10
        else:
            new += n
            break
    return new

dic = [0 for _ in range(10001)]
dic[0] = 1
for i in range(1, 10001):
    next_num = make_num(i)
    if next_num <= 10000:
        dic[next_num] = 1

for i in range(10001):
    if dic[i] == 0:
        print(i)

File: node.js/백준/Silver/4673. 셀프 넘버/셀프 넘버.js
const makeNumber = (n) => {
    let newNum = n;
    while (true) {
        if (Math.floor(n / 10) >= 1) {
            newNum += n % 10
            n = Math.floor(n/10)
        } else {
            newNum += n
            break
        }
    }
    return newNum
}

const dic = Array.from({length: 10001}, () => 0)
dic[0] = 1
for (i=1; i<=10001; i++) {
    const nextNum = makeNumber(i)
    if (nextNum <= 10000) {
        dic[nextNum] = 1
    }
}

dic.forEach((ele, idx) => {
    if (ele == 0) {
        console.log(idx)
    }
})